pylearn2/models/mlp.py

```diff
@@ -3059,6 +3079,14 @@ def set_input_space(self, space):
 
         rng = self.mlp.rng
 
+        if self.pad != (0, 0):
+            output_shape = \
+                [int(np.ceil((i_sh + 2. * k_pad - k_sh) / float(k_st))) + 1
+                 for i_sh, k_sh, k_st, k_pad in izip(self.input_space.shape,
+                                                     self.kernel_shape,
+                                                     self.kernel_stride,
+                                                     self.pad)]
+
         if self.border_mode == 'valid':
```

Comment:
We should make sure 'border_mode' and 'pad' are not passed at the same time, I guess.
Also, the `output_shape` will be overwritten by default.